common/utils.js

```diff
@@ -1339,6 +1339,33 @@
             }
         }
 
+        function overrideDownloadClickBehavior(containerElem) {
+            var downloadElements = containerElem[0].querySelectorAll(".deriva-url-validate");
+
+            downloadElements.forEach(function (el) {
```

Comment:
I tested this on IE11. It's throwing an error here. It seems like `forEach` is not defined for the returned value of `querySelectorAll` in IE implementation according to this document:
https://developer.mozilla.org/en-US/docs/Web/API/NodeList

You could either change it to simple for loop, or use the suggested method in that document which is the following:

```javascript
Array.prototype.forEach.call(downloadElements, function (el) {
  // the function goes here
});
```

Reply:
You should also test this on Firefox. It's not registering the click handler in the recordset app. It seems like it's calling this function before rendering the table rows. It works fine in record page.

Reply:
About the `forEach`, you actually don't need it. you can just attach the event handler on the result of querySelectorAll.

```javascript
angular.element(downloadElements).on('click', function (e) {

});
```